astropy/cosmology/core.py

```diff
@@ -2964,6 +3181,22 @@ def vectorize_if_needed(func, *x):
         return func(*x)
 
 
+def inf_like(x):
```

Comment:
@mhvk - This points beyond this PR, but I wonder whether it make sense to move these helper functions outside of `cosmology` to astropy.utils? (and ultimately propagate them upstream to numpy?)

Reply:
I actually just realized why what @aconley and I came up with here isn't exactly what we want here, for reasons that relate to why `numpy` doesn't have an `inf_like` function.  The `np.*_like` functions preserve both shape and type.  But in this case we don't actually want to do that.  We want to promote to float.  There is no `inf` in an integer type.

Reply:
However, within `astropy` I imagine that what we want is to always promote from integer to float when a result would trigger an infinity.  While we want to except a variety of user inputs, we are generically returning floats for most calculations (coordinates in sexagesimal being the obvious exception that occurs first to me).  So thus `astropy.utils.inf_like` might be a reasonable thing to exist.

Reply:
I think this is insufficiently generally useful to move it elsewhere -- wrapping `np.full_like` would seem just the right thing to do.

p.s. I think it is quite nice to have documentation even on private functions; it is still useful for other developers!